src/lib/http-agent.js

```diff
@@ -0,0 +1,77 @@
+const { URL } = require('url')
+const { NETLIFYDEVERR, NETLIFYDEVWARN } = require('../utils/logo')
+const waitPort = require('wait-port')
+const { HttpsProxyAgent } = require('https-proxy-agent')
+const fs = require('fs-extra')
+
+// https://github.com/TooTallNate/node-https-proxy-agent/issues/89
+class HttpsProxyAgentWithCA extends HttpsProxyAgent {
+  constructor(opts) {
+    super(opts)
+    this.ca = opts.ca
+  }
+
+  callback(req, opts) {
+    return super.callback(req, {
+      ...opts,
+      ...(this.ca && { ca: this.ca }),
+    })
+  }
+}
+
+const getAgent = async ({ httpProxy, certificateFile, log, exit }) => {
+  if (!httpProxy) {
+    return undefined
+  }
+
+  let proxyUrl
+  try {
+    proxyUrl = new URL(httpProxy)
+  } catch (e) {
+    log(NETLIFYDEVERR, `${httpProxy} is not a valid URL`)
+    exit(1)
+  }
+
+  const scheme = proxyUrl.protocol.substr(0, proxyUrl.protocol.length - 1)
+  if (!['http', 'https'].includes(scheme)) {
+    log(NETLIFYDEVERR, `${httpProxy} must have a scheme of http or https`)
+    exit(1)
+  }
+
+  try {
+    const open = await waitPort({
+      port: parseInt(proxyUrl.port) || (scheme === 'http' ? 80 : 443),
+      host: proxyUrl.hostname,
+      timeout: 50,
+      output: 'silent',
+    })
+    if (!open) {
+      throw new Error(`Can't connect to '${httpProxy}'`)
+    }
+  } catch (e) {
+    log(NETLIFYDEVERR, `${httpProxy} is not available`)
+    exit(1)
+  }
+
+  let certificate
+  if (certificateFile) {
+    try {
+      certificate = await fs.readFile(certificateFile)
+    } catch (error) {
+      log(NETLIFYDEVWARN, `could not read certificate file '${certificateFile}'`)
```

Comment:
Should `error` be printed here as well? To get some additional context: permission error, does not exist, etc.

Reply:
Good point, I'll fix it

Reply:
Fixed in https://github.com/netlify/cli/pull/1215/commits/7a41098f5c04bcf918f3c42e70556ecb4d1c8b56#diff-4c26db14dfb469fcc316ade4b76309c0R66